### PR TITLE
Chung

### PR DESCRIPTION
**Pull request by ChungNguyenVr912**
1. Refactor...
2. Fix chat messages load
3. Feature: chat typing status
4. Fix: personal page, friends dropdown menu
5. Feature: bottom chat and right chat (old messages)

### CODE REVIEW

File: src/App.jsx [hunk 1]
```diff
@@ -39,20 +42,33 @@ function App() {
             console.log(frame)
             SocketClient.subscribe('/users/queue/messages', (message) => {
                 const messageContent = JSON.parse(message.body);
-                console.log(messageContent)
-                if (messageContent && messageContent.sender !== user.email){
-                    const notify = new Notification(messageContent.senderName, {
-                        icon: "src/assets/img/logo.svg",
-                        body: messageContent.content,
-                    });
+                // if (messageContent && messageContent.sender !== user.email) {
```
Comment (daonguyendev) [inline src/App.jsx:45]: Remove redundant code if not used anymore. But if will use it for future features, should comment: //TODO: ... to declare your code block usage

File: src/App.jsx [hunk 2]
```diff
@@ -39,20 +42,33 @@ function App() {
...
+                        user: messageContent.sender,
+                        typingStatus: messageContent.typingStatus
+                    }))
                 }
-                dispatch(addNewMessage(messageContent))
             })
             SocketClient.subscribe('/users/queue/notify', (message) => {
-                const messageContent = message.body;
-                console.log(messageContent)
+                const notify = message.body;
```
Comment (daonguyendev) [inline src/App.jsx:69]: Using edit to notification (Noun) for variable instead of notify (Verb)

File: src/api/accountApi.js [hunk 3]
```diff
@@ -18,7 +17,7 @@ export const registerApi = async (data) => {
     try {
         response = await axios.post(VIBELY_API + '/users', data)
     } catch (e) {
-        console.log("Register error: " + e);
+        console.log(e)
```
Comment (daonguyendev) [inline src/api/accountApi.js:20]: What is the type of error?

File: src/api/friendListApi.js [hunk 4]
```diff
@@ -12,8 +12,7 @@ export const findFriends = async () => {
             }
         });
     } catch (e) {
-        console.log(e.response);
-        return e.response
+        console.log(e);
```
Comment (daonguyendev) [inline src/api/friendListApi.js:15]: What is the type of error?

File: src/assets/scss/components/_chat.scss [hunk 5]
```diff
@@ -307,7 +334,9 @@
     -moz-box-align: start;
     -ms-flex-align: start;
     align-items: flex-start;
-    padding-bottom: 70px !important;
+    //padding-bottom: 70px !important;
```
Comment (daonguyendev) [inline src/assets/scss/components/_chat.scss:337]: Remove redundant codes

File: src/assets/scss/components/_navigation.scss [hunk 6]
```diff
@@ -52,7 +52,7 @@
 
   @include media-breakpoint-down(lg) {
     padding: 9px 15px 9px 15px;
-    height: auto;
+    //height: auto;
```
Comment (daonguyendev) [inline src/assets/scss/components/_navigation.scss:55]: Remove redundant codes

File: src/components/ChatBox/index.jsx [hunk 7]
```diff
@@ -1,97 +1,153 @@
...
+            }))
         }
     }, [currentConversation, page])
 
+// reset page
     useEffect(() => {
         setPage(0)
+        dispatch(resetNewMessages())
+        chatInput.current.focus()
+        // setTimeout(()=>{
```
Comment (daonguyendev) [inline src/components/ChatBox/index.jsx:74]: Remove redundant codes

**Review by daonguyendev — COMMENTED** [review]
Please review my comments below and clean your code

File: src/App.jsx (hunk 1 above)
Comment (ChungNguyenVr912) [reply src/App.jsx:45]: Resolved

File: src/App.jsx (hunk 2 above)
Comment (ChungNguyenVr912) [reply src/App.jsx:69]: Resolved

File: src/api/accountApi.js (hunk 3 above)
Comment (ChungNguyenVr912) [reply src/api/accountApi.js:20]: Resolved

File: src/api/friendListApi.js (hunk 4 above)
Comment (ChungNguyenVr912) [reply src/api/friendListApi.js:15]: Resolved

File: src/assets/scss/components/_navigation.scss (hunk 6 above)
Comment (ChungNguyenVr912) [reply src/assets/scss/components/_navigation.scss:55]: Resolved

File: src/components/ChatBox/index.jsx (hunk 7 above)
Comment (ChungNguyenVr912) [reply src/components/ChatBox/index.jsx:74]: Resolved

**Review by daonguyendev — APPROVED** [review]

**Review by thaole3107 — APPROVED** [review]

**Review by HuyVu18 — APPROVED** [review]

**Review by MinhTam1509 — APPROVED** [review]

**Pull request merged**